app/ModelFunctions/AlbumsFunctions.php

```diff
@@ -72,9 +72,11 @@ public function prepare_albums(BaseCollection $albums, BaseCollection $children)
 				$album_array['owner'] = $albums[$key]->owner->username;
 			}
 
-			$thumbs = $this->albumFunctions->get_thumbs($albums[$key], $children[$key]);
-			$this->albumFunctions->set_thumbs($album_array, $thumbs);
-			$this->albumFunctions->set_thumbs_children($album_array['albums'], $thumbs[1]);
+			if ($this->readAccessFunctions->album($albums[$key]) === 1) {
```

Comment:
I actually liked the thumb leakage.
Removing this if check enables me to have both a thumb an an password protected album.

I know my demand is quite special.
Can we add an 'advanced setting' that allows for this use-case?

Thanks!

Reply:
LOL. If ever https://xkcd.com/1172/ was more true :wink:...